test/framework/src/main/java/org/elasticsearch/index/mapper/MapperTestCase.java

```diff
@@ -53,6 +62,50 @@
 public abstract class MapperTestCase extends MapperServiceTestCase {
     protected abstract void minimalMapping(XContentBuilder b) throws IOException;
 
+    protected void field(XContentBuilder builder) throws IOException {
+        builder.field("field");
+        fieldValue(builder);
+    }
+
+    protected abstract void fieldValue(XContentBuilder builder) throws IOException;
+
+    public final void testExistsQuery() throws IOException {
+        MapperService mapperService = createMapperService(fieldMapping(this::minimalMapping));
+        ParsedDocument doc = mapperService.documentMapper().parse(source(this::field));
+        withLuceneIndex(mapperService, iw -> iw.addDocument(doc.rootDoc()), reader -> {
```

Comment:
I don't think you need to index the document, you can just look at the fields in the parsed doc?

Reply:
good point, thanks for raising this, will fix